.github/workflows/delete-merged-branch.yml

```
@@ -0,0 +1,48 @@
+          script: |
+            const pr = context.payload.pull_request;
+            const branchName = pr.head.ref;
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            
+            // main ブランチは削除しない
+            if (branchName === 'main') {
+              console.log(`Skipping deletion of protected branch: ${branchName}`);
+              return;
+            }
+            
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)